pkg/sync/store_wrapper.go

```diff
@@ -0,0 +1,100 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/celestiaorg/go-header"
+
+	syncnotifier "github.com/evstack/ev-node/pkg/sync/notifier"
+)
+
+type publishFn[H header.Header[H]] func([]H)
+
+type instrumentedStore[H header.Header[H]] struct {
+	delegate header.Store[H]
```

Comment:
You could just use `header.Store[H]` here and only overwrite `Append` to save some lines. This can make it easier to stay compatible with the underlying store in the long run